source/chapter3-debian.rst

```diff
@@ -5,3 +5,176 @@ Debian
+    cd -
+    sudo umount /mnt
+
+
+Create docker image:
+-------------------
+For arm64
+
+.. code-block:: bash
+
+    docker build --force-rm --file Dockerfile.arm64 --tag creation .
+For armhf
+
+.. code-block:: bash
+
+    docker build --force-rm --file Dockerfile.armhf --tag creation .
+
+Execute docker image on interactive mode with priviledge:
+--------------------------------------------------------
+
+.. code-block:: bash
+
+    docker run -it --privileged -v $PWD:/home/build creation
+
+The current directory are shared with docker.
+
+
+
+Create Debian image:
+--------------------
+On Docker environment and fai directory
+
+.. code-block:: bash
+
+    DOCKER> ./build-fai-linaro-debian.sh
+    DOCKER> exit
+
+Image generation per board
+==========================
+For qemu (armhf):
+-----------------
+
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/ledge-stm32mp157c-dk2/FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template -O  FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template
+
+    (./script/create_raw_from_flashlayout.sh FlashLayout_sdcard_armhf_without_boot_firmware.fld  &&
+        pigz -9 FlashLayout_sdcard_armhf_without_boot_firmware.raw)
+
+For qemu (arm64):
+-----------------
+
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/aarch64/FlashLayout_sdcard_arm64_without_boot_firmware.fld -O  FlashLayout_sdcard_arm64_without_boot_firmware.fld
+
+    (./script/create_raw_from_flashlayout.sh FlashLayout_sdcard_arm64_without_boot_firmware.fld &&
+        pigz -9 FlashLayout_sdcard_arm64_without_boot_firmware.raw)
+
+For stm32mp157c-dk2 (armhf):
+-----------------
+
+.. code-block:: bash
+
+    mkdir -p script
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/raw-tools/create_raw_from_flashlayout.sh \
+     -O script/create_raw_from_flashlayout.sh
+    chmod +x script/create_raw_from_flashlayout.sh
+
+    wget https://raw.githubusercontent.com/Linaro/meta-ledge/zeus/meta-ledge-bsp/recipes-devtools/generate-raw-image/files/ledge-stm32mp157c-dk2/FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template -O FlashLayout_sdcard_ledge-stm32mp157c-dk2-debian.tsv.template
```

Comment:
shall this template be changed with specifying %%BOOTFS% ?

Reply:
The script and flashlayout must be aligned for the image filtering

Reply:
then it's better to mention about that in the doc.

Reply:
yes, but the user cannot see this specificity, it's the script which manage it

Reply:
For debian the %%BOOTFS%% are not used, the bootfs are linked to image generated by oe qemu build

Reply:
can you please add here what directory structure has to be for this script?

Reply:
Sorry, I don't understand what you want ? The script need to have the fld and the debian tarball and for all others files specified on fld which are not present, it dowload it (it the same for qemu)

Reply:
ah, ok. In case of download did you found a way to pass linaro authorization?

Reply:
No, the credential of linaro are two complicated to be by-passed